### PR TITLE
Addressing feedback from code review:

### PR DESCRIPTION
**Pull request by Ivanca**
- Creating .in file for pip-compile (to generate requirements.txt)
- Moving pytest.ini settings to setup.cfg (removes pytest.ini)
- Uncomment on Makefile 'clean' : rf -rf *.egg-info
- Moving coverage.rc settings to setup.cfg (removes coverage.rc)
- test_utils no longer used @edx.org domain (uses example.com)
- README.md improved with usage instructions and better install instructions

### CODE REVIEW

**Comment by felipemontoya** [comment]
@diegomillan @Squirrel18 @andrey-canon I asked Ivan to open the PR again and wait for your feedback here.

**Comment by Ivanca** [comment]
> @diegomillan @Squirrel18 @andrey-canon I asked Ivan to open the PR again and wait for your feedback here.

There is a chance they don't have further feedback about it... 😐

**Comment by diegomillan** [comment]
HI @Ivanca , I will try to review this today

File: README.md [hunk 1]
```diff
@@ -19,19 +41,29 @@ make dev.clone
 make dev.provision
 make dev.provision
```
Comment (Squirrel18) [inline README.md]: Command twice.

File: eox_core/api/v1/urls.py [hunk 2]
```diff
@@ -1,11 +1,9 @@
...
 
 from django.conf.urls import url
-from rest_framework_jwt.views import obtain_jwt_token
 from eox_core.api.v1 import views
 
 urlpatterns = [  # pylint: disable=invalid-name
-    url(r'^api-token-auth/', obtain_jwt_token),
     url(r'^user/$', views.EdxappUser.as_view(), name='edxapp-user'),
     url(r'^enrollment/$', views.EdxappEnrollment.as_view()),
     url(r'^userinfo$', views.UserInfo.as_view()),
```
Comment (Squirrel18) [inline eox_core/api/v1/urls.py]: i think it should be set the name value to each url definition, just for a better reverse operation.

File: eox_core/api/v1/views.py [hunk 3]
```diff
@@ -116,7 +116,7 @@ class UserInfo(APIView):
     Auth-only view to check some basic info about the current user
     Can use Oauth2/Session/JWT authentication
```
Comment (andrey-canon) [inline eox_core/api/v1/views.py]: Update docstring, JWT authentication was removed

File: eox_core/api/v1/urls.py (hunk 2 above)
Comment (andrey-canon) [inline eox_core/api/v1/urls.py]: missing backslash

Comment (andrey-canon) [inline eox_core/api/v1/urls.py]: missing backslash at end of regex expression

Comment (Ivanca) [reply eox_core/api/v1/urls.py]: Fixed in latest commit.

File: eox_core/api/v1/views.py (hunk 3 above)
Comment (Ivanca) [reply eox_core/api/v1/views.py]: Fixed in latest commit.

File: eox_core/api/v1/urls.py (hunk 2 above)
Comment (Ivanca) [reply eox_core/api/v1/urls.py]: Fixed in latest commit.

File: README.md (hunk 1 above)
Comment (Ivanca) [reply README.md]: Fixed in latest commit.

**Pull request merged**